tardis/tests/test_rpacket.py

```diff
@@ -0,0 +1,146 @@
+	assert tests.test_rpacket_get_r(double_value)
+
+np.random.seed(5)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_tau_event(double_value):
+	assert tests.test_rpacket_get_tau_event(double_value)
+
+np.random.seed(6)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_nu_line(double_value):
+	assert tests.test_rpacket_get_nu_line(double_value)
+
+np.random.seed(7)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_nu_line(double_value):
+	assert tests.test_rpacket_get_d_boundary(double_value)
+
+np.random.seed(8)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_nu_line(double_value):
+	assert tests.test_rpacket_get_d_electron(double_value)
+
+np.random.seed(9)
+@pytest.mark.parametrize(
+	"double_value", map(c_double, np.random.uniform(
+						MIN_FLOAT, MAX_FLOAT, size=SIZE))
+	)
+def test_rpacket_get_nu_line(double_value):
+	assert tests.test_rpacket_get_d_line(double_value)
+
+
+# Testing functions with Integer valued parameters
+np.random.seed(10)
+@pytest.mark.parametrize(
+	"unsigned_int_value", np.random.randint(0, MAX_INT, size=SIZE)
+	)
+def test_rpacket_get_current_shell_id(unsigned_int_value):
+	assert tests.test_rpacket_get_current_shell_id(unsigned_int_value)
+
+np.random.seed(11)
+@pytest.mark.parametrize(
+	"nt_value", np.random.randint(MIN_INT, MAX_INT, size=SIZE)
+	)
+def test_rpacket_get_next_line_id(int_value):
+	assert tests.test_rpacket_get_next_line_id(int_value)
+
+np.random.seed(12)
+@pytest.mark.parametrize(
+	"int_value", np.random.randint(MIN_INT, MAX_INT, size=SIZE)
+	)
+def test_rpacket_get_next_line_id(int_value):
+	assert tests.test_rpacket_get_recently_crossed_boundary(int_value)
+
+np.random.seed(13)
+@pytest.mark.parametrize(
+	"int_value", np.random.randint(MIN_INT, MAX_INT, size=SIZE)
+	)
+def test_rpacket_get_next_line_id(int_value):
+	assert tests.test_rpacket_get_virtual_packet_flag(int_value)
+
+np.random.seed(14)
+@pytest.mark.parametrize(
+	"int_value", np.random.randint(MIN_INT, MAX_INT, size=SIZE)
+	)
+def test_rpacket_get_next_line_id(int_value):
+	assert tests.test_rpacket_get_virtual_packet(int_value)
+
+np.random.seed(15)
+@pytest.mark.parametrize(
+	"int_value", np.random.randint(MIN_INT, MAX_INT, size=SIZE)
+	)
+def test_rpacket_get_next_line_id(int_value):
+	assert tests.test_rpacket_get_next_line_id(int_value)
+
+
+# Testing functions without any parameters
+def test_rpacket_get_last_line():
+	assert tests.test_rpacket_get_last_line()
+
+def test_rpacket_get_close_line():
+	assert tests.test_rpacket_get_close_line()
+
+def test_rpacket_get_status():
+	assert tests.test_rpacket_get_status()
```

Comment:
This part is a little tricky, I'll write the tests for this along with the ones in `cmontecarlo.c`.